src/ints/bios_disk.cpp

```diff
@@ -164,10 +164,10 @@ uint8_t imageDisk::Read_Sector(uint32_t head,uint32_t cylinder,uint32_t sector,v
 
 uint8_t imageDisk::Read_AbsoluteSector(uint32_t sectnum, void *data)
 {
-	const uint32_t bytenum = sectnum * sector_size;
+	const cross_off_t bytenum = static_cast<cross_off_t>(sectnum) * sector_size;
```

Comment:
Is it possible to use `auto` here and let the RHS define the type?

(Although might be needed if the `sector-size` type collapses the result to a signed value - if possible, avoid any collapsing back down to `int` or signed, to avoid truncating for the assignment)